src/core/adapters/SimplePoolAdapter.ts

```diff
@@ -136,14 +141,19 @@ export abstract class SimplePoolAdapter implements IProtocolAdapter {
     toBlock,
   }: GetEventsInput): Promise<MovementsByBlock[]> {
     return await this.getMovements({
-      protocolTokenAddress,
+      protocolToken: await this.fetchProtocolTokenMetadata(
+        protocolTokenAddress,
+      ),
       underlyingTokens: await this.fetchUnderlyingTokensMetadata(
         protocolTokenAddress,
       ),
-      fromBlock,
-      toBlock,
-      from: userAddress,
-      to: undefined,
+      filter: {
+        smartContractAddress: protocolTokenAddress,
```

Comment:
I like this solution for now, but we need to keep an eye if there's a pattern for other staking adapters in which the filter address is different than the protocol token address, as then we might want to create a new template adapter.